source/common/upstream/upstream_impl.cc

```diff
@@ -307,11 +306,14 @@ ClusterInfoImpl::ClusterInfoImpl(const envoy::api::v2::Cluster& config,
     }
   }
 
+  Server::Configuration::TransportSocketFactoryContextImpl factory_context(
+      ssl_context_manager, *stats_scope_, secret_manager, init_manager);
   auto& config_factory = Config::Utility::getAndCheckFactory<
       Server::Configuration::UpstreamTransportSocketConfigFactory>(transport_socket.name());
   ProtobufTypes::MessagePtr message =
       Config::Utility::translateToFactoryConfig(transport_socket, config_factory);
-  transport_socket_factory_ = config_factory.createTransportSocketFactory(*message, *this);
+  transport_socket_factory_ =
```

Comment:
Do we even need to this here? Can we create the transport socket factory outside of the constructor and just pass it in for storage? This would further simplify the constructor args, etc.